### PR TITLE
Buffs H.U.E.Y upgrades (plus advanced mk2 weedkill one)

### LINKED ISSUE #37242: make H.U.E.Y. service borgs not shit
(WEB REPORT BY: jesture REMOTE: 172.93.101.77:7777)
give them a way to remove dead plants from trays

when you try to wipe the data from a gene disk gives you the prompt to wipe them but doesn't actually wipe them

also you can't even grab data disks from diskette boxes

also give them a spade

also maybe a water tank you can dispense from that recharges might be a good addition as well

okay that's all i can think of right off the top of my head

### PR DESCRIPTION
**Pull request by SECBATON-GRIFFON**
[content][qol]

## What this does
Closes #37242.
From the requests on the upgrade:
Makes the gripper in this upgrade able to hold botany disks grabbed out of diskette boxes
Allows them to properly wipe disks
Gives them a spade, allows spades to remove dead plants when used on them. (covering both at once)
Gives them a water bottle that refills on recharge

Additionally:
Gives them a hatchet for logs (dunno about this)
Gives them a recharging eznutrient bottle
Also unsure about giving them a bottle of recharging insecticide, either in the first or second upgrade. What's the feedback on this?

And then gives them another "MK2" upgrade that needs biology 5, engineering 4 and data 4, focused on vine killing and mutating, gives them a floral somatoray, scythe and plantbgone spray. The spray automatically refills. (maybe some beekeeping stuff too like a gripper for apiaries, a net and bee stuff? also is the somatoray too much?)

On the backend, standardises all the spray and borg bottle autofill code

## Why it's good
makes botany more doable with this upgrade

## How it was tested
applying the upgrades, grabbing botany disks out of the box, wiping them after putting seed data on them, planting grass, using plantbgone, removing it with the spade, cutting a log with the hatchet

## Changelog

:cl:
 * rscadd: Service cyborg H.U.E.Y upgrades now contain a spade, hatchet and recharging bottles of water and EZ nutrient.
 * rscadd: Service cyborgs can now get a MK2 H.U.E.Y upgrade from research, for biology 5, engineering 4 and data 4. Contains a floral somatoray, scythe and plant-b-gone spray.
 * tweak: Spades can now remove dead plants when used on it, to help borgs with them.
 * bugfix: H.U.E.Y borg service grippers can now wipe botany disks and take them out of diskette boxes.

### CODE REVIEW

**Comment by jwhitak** [comment]
I actually was working on this and was going to drop a PR likely tomorrow evening when I'm off of work. RIP

**Comment by SECBATON-GRIFFON** [comment]
> I actually was working on this and was going to drop a PR likely tomorrow evening when I'm off of work. RIP

what were you gonna add to the borgs?

**Comment by jwhitak** [comment]
Less than what you did, so it's fine.

I'm concerned by the combination of an infinite water spray and a hatchet. A stun and a lethal weapon... Probably shouldn't give them a hatchet directly. If you do want them to be able to cut wood, check out the flags for things that can chop wood. Give them something new and custom like a safety saw that can't harm people normally. (Edit: see grug #36916 for more info)

File: code/modules/reagents/reagent_containers/glass/bottle/robot.dm [hunk 1]
```diff
@@ -17,20 +21,23 @@
...
-
 /obj/item/weapon/reagent_containers/glass/bottle/robot/antitoxin
 	name = "internal anti-toxin bottle"
 	desc = "A small bottle of Anti-toxins. Counters poisons, and repairs damage, a wonder drug."
 	icon = 'icons/obj/chemical.dmi'
 	//icon_state = "bottle17"
 	reagent = ANTI_TOXIN
+	
+/obj/item/weapon/reagent_containers/glass/bottle/robot/water
+	name = "internal water bottle"
```
Comment (jwhitak) [inline code/modules/reagents/reagent_containers/glass/bottle/robot.dm:32]: No time for a full review but can you break this over someone's head, permanently damaging it? (Thanks @Dacendeth )

**Comment by cloutcode** [comment]
complainingchads... we did it...

File: code/modules/reagents/reagent_containers/glass/bottle/robot.dm (hunk 1 above)
Comment (SECBATON-GRIFFON) [reply code/modules/reagents/reagent_containers/glass/bottle/robot.dm:32]: you'd have to ask the medborgs that too! but seemingly not (that code is in food/drinks subtypes)

**Comment by SECBATON-GRIFFON** [comment]
> I'm concerned by the combination of an infinite water spray

bottle, plus getting more requires going to a recharge station

> and a hatchet

only has 2 more force than the kitchen knife they come with, and 0.3 less sharpness too. more throwforce though, but borgs can't exactly throw their modules anyways.

**Pull request merged**